lib/ansible/module_utils/urls.py

```diff
@@ -864,47 +913,40 @@ def detect_no_proxy(self, url):
                     return False
         return True
 
-    def _make_context(self, to_add_ca_cert_path):
+    def make_context(self, cafile, cadata):
+        cafile = self.ca_path or cafile
+        if self.ca_path:
```

Comment:
Is this code working as expected?

`maybe_add_ssl_handler(url, validate_certs, ca_path=ca_path)` creates a `SSLValidationHandler` with a possible not-`None` `ca_path`.

If `ca_path` within the handler is not `None`, the `cadata` will be created and returned by `get_ca_certs()` and but then when its passed in here it won't be used, so what's the point in creating the `cadata` at all?

I ask because I'm having TLS validation issues with 2.7.0, 2.8.1, 2.8.2, and `devel` on MacOS Mojave.

I was thinking about contributing a feature where if `certifi` is installed, it would use that for the `ca_path` by default:

```
try:
    import certifi
    HAS_CERTIFI = True
except ImportError:
    HAS_CERTIFI = False

# ...

    def __init__(self, hostname, port, ca_path=None):
        self.hostname = hostname
        self.port = port
        self.ca_path = ca_path

        if self.ca_path is None and HAS_CERTIFI
            self.ca_path = certifi.where()
```

but `ssl.PEM_cert_to_DER_cert` doesn't like the comments in the cacert.pem file `certifi` provides. Also if you look at the source for that method:

```
def PEM_cert_to_DER_cert(pem_cert_string):
    """Takes a certificate in ASCII PEM format and returns the
    DER-encoded version of it as a byte sequence"""

    if not pem_cert_string.startswith(PEM_HEADER):
        raise ValueError("Invalid PEM encoding; must start with %s"
                         % PEM_HEADER)
    if not pem_cert_string.strip().endswith(PEM_FOOTER):
        raise ValueError("Invalid PEM encoding; must end with %s"
                         % PEM_FOOTER)
    d = pem_cert_string.strip()[len(PEM_HEADER):-len(PEM_FOOTER)]
    return base64.decodestring(d.encode('ASCII', 'strict'))
```

it only rips out the beginning and ending header, so it doesn't seem like it was designed for CA bundles but my code works which had me questioning if `cadata` is ever even used.